local/iatlas/cbioportal_export/maf_to_cbioportal.py

```diff
@@ -0,0 +1,347 @@
+
+    Returns:
+        Dict[str, pd.DataFrame]: A dictionary with the following keys:
+            "annotated_maf" : the annotated maf dataset concatenated together
+            "error_maf" : error report outputted by genome nexus of failed annotations
+    """
+    dataset_dir = os.path.join(
+        f"{datahub_tools_path}/add-clinical-header/", dataset_name
+    )
+    annotated_mafs = []
+    error_mafs = []
+    for i in range(1, n_maf_chunks + 1):
+        annotated_mafs.append(
+            pd.read_csv(
+                f"{dataset_dir}/data_mutations_annotated_{i}.txt", sep="\t", comment="#"
+            )
+        )
+        error_mafs.append(
+            pd.read_csv(f"{dataset_dir}/data_mutations_error_report_{i}.txt", sep="\t")
+        )
+
+    annotated_mafs_all = pd.concat(annotated_mafs)
+    error_mafs_all = pd.concat(error_mafs)
+
+    annotated_mafs_all_processed = postprocessing(input_df=annotated_mafs_all)
+    annotated_mafs_text = utils.remove_pandas_float(df = annotated_mafs_all_processed)
+    
+    with open(f"{dataset_dir}/data_mutations.txt", "w") as new_maf_f:
+        new_maf_f.write(annotated_mafs_text)
+        
+    error_mafs_all.to_csv(
+        f"{dataset_dir}/data_mutations_error_report.txt", sep="\t", index=False
+    )
+    return {"annotated_maf": annotated_mafs_all, "error_maf": error_mafs_all}
+
+
+def postprocessing(input_df: pd.DataFrame) -> pd.DataFrame:
+    """Additional postprocessing steps we have to do for the
+       datasets. This can be removing variants that
+       cannot be annotated, etc.
+
+    Args:
+        input_df (pd.DataFrame): input annotated maf file
+
+    Returns:
+        pd.DataFrame: output maf file postprocessed
+    """
+    annotated_maf = input_df.copy()
+    # remove the failed variants on MT chromosome
+    annotated_maf = annotated_maf[annotated_maf.Chromosome != "chrM"]
+    return annotated_maf
+
+
+def save_to_synapse(
+    dataset_name: str,
+    datahub_tools_path: str,
+    output_folder_synid: str,
+    version_comment: str = None,
+) -> None:
+    """Saves the dataset's annotated maf file and error report file
+        to synapse
+
+    Args:
+        dataset_name (str): name of the iatlas dataset to save to
+            synapse
+        datahub_tools_path (str): Path to the datahub tools repo
+        output_folder_synid (str): Synapse id of the output folder
+        version_comment (str): Version comment for this iteration of files on synapse. Optional.
+            Defaults to None.
+    """
+    dataset_dir = os.path.join(datahub_tools_path, "add-clinical-header", dataset_name)
+    # see if dataset_folder exists
+    dataset_folder_exists = False
+    for _, directory_names, _ in synapseutils.walk(syn=syn, synId=output_folder_synid):
+        directories = directory_names  # top level directories
+        break
+
+    for dataset_folder in directories:
+        if dataset_name == dataset_folder[0]:
```

Comment:
Should this be case insensitive?

Reply:
Hmm the dataset names appear in a specific format in the datasets and that's how it's outputted / dataset specific folders are created and our collaborators already expect it in that format, so decided to be case sensitive here